File: apps/desktop/src/types/job.ts
export type JobStatus =
  "queued" | "running" | "paused" | "completed" | "failed" | "cancelled";

export interface JobOut {
  job_id: string;
  project_id: string | null;
  type: string;
  status: JobStatus;
  progress: number;
  model_id: string;
  provider_id: string;
  capability: string;
  error: string | null;
  error_category: string;
  attempts: number;
  result: Record<string, unknown> | null;
  batch_id: string;
  batch_label: string;
  target_id: string;
  target_label: string;
  has_remote_task: boolean;
  created_at: string;
  started_at: string | null;
  completed_at: string | null;
  paused_at: string | null;
  cancelled_at: string | null;
}

export interface ProjectJobState {
  project_id: string;
  paused: boolean;
}

File: apps/backend/app/schemas/job.py
"""Phase 10 — 统一任务中心 Schema（/api/jobs）。"""

from datetime import datetime
from typing import Literal

from pydantic import BaseModel, Field


JobStatusLiteral = Literal[
    "queued", "running", "paused", "completed", "failed", "cancelled"
]


class JobOut(BaseModel):
    job_id: str
    project_id: str | None = None
    type: str
    status: JobStatusLiteral
    progress: int = Field(default=0, ge=0, le=100)
    model_id: str = ""
    provider_id: str = ""
    capability: str = ""
    error: str | None = None
    error_category: str = ""
    attempts: int = 0
    result: dict | None = None
    created_at: datetime
    started_at: datetime | None = None
    completed_at: datetime | None = None
    paused_at: datetime | None = None
    cancelled_at: datetime | None = None


class BatchJobsRequest(BaseModel):
    project_id: str = Field(min_length=1, max_length=100)
    action: Literal["cancel", "pause", "resume"]
    stage: str | None = Field(default=None, max_length=50)


class BatchJobsResult(BaseModel):
    affected: int = 0
    jobs: list[JobOut] = Field(default_factory=list)
